pkg/syncer/shared/namespace.go

```diff
@@ -20,40 +20,75 @@ import (
 	"crypto/sha256"
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/kcp-dev/logicalcluster"
+	"github.com/martinlindhe/base36"
 )
 
 const (
-	NamespaceLocatorAnnotation = "kcp.dev/namespace-locator"
+	NamespaceAnnotation       = "workload.kcp.dev/namespace"
+	WorkspaceAnnotation       = "workload.kcp.dev/workspace"
+	SyncerWorkspaceAnnotation = "workload.kcp.dev/syncer"
 )
 
 // NamespaceLocator stores a logical cluster and namespace and is used
 // as the source for the mapped namespace name in a physical cluster.
 type NamespaceLocator struct {
-	LogicalCluster logicalcluster.Name `json:"logical-cluster"`
-	Namespace      string              `json:"namespace"`
+	Workspace       logicalcluster.Name `json:"logical-cluster"`
+	SyncerWorkspace logicalcluster.Name `json:"syncer-workspace"`
+	Namespace       string              `json:"namespace"`
 }
 
 func LocatorFromAnnotations(annotations map[string]string) (*NamespaceLocator, error) {
-	annotation := annotations[NamespaceLocatorAnnotation]
-	if len(annotation) == 0 {
+	if len(annotations) == 0 {
 		return nil, nil
 	}
+
 	var locator NamespaceLocator
-	if err := json.Unmarshal([]byte(annotation), &locator); err != nil {
-		return nil, err
+	if val, ok := annotations[NamespaceAnnotation]; ok && val != "" {
+		locator.Namespace = val
+	} else {
+		return nil, fmt.Errorf("namespace annotation not found or invalid")
+	}
+
+	if val, ok := annotations[WorkspaceAnnotation]; ok && val != "" {
+		locator.Workspace = logicalcluster.New(val)
+	} else {
+		return nil, fmt.Errorf("workspace annotation not found or invalid")
+	}
+
+	if val, ok := annotations[SyncerWorkspaceAnnotation]; ok && val != "" {
+		locator.SyncerWorkspace = logicalcluster.New(val)
+	} else {
+		return nil, fmt.Errorf("syncer workspace annotation not found or invalid")
 	}
+
 	return &locator, nil
 }
 
+func AnnotationsFromLocator(l NamespaceLocator) map[string]string {
+	annotations := make(map[string]string)
+	annotations[NamespaceAnnotation] = l.Namespace
+	annotations[WorkspaceAnnotation] = l.Workspace.String()
+	annotations[SyncerWorkspaceAnnotation] = l.SyncerWorkspace.String()
+	return annotations
+}
+
 // PhysicalClusterNamespaceName encodes the NamespaceLocator to a new
 // namespace name for use on a physical cluster. The encoding is repeatable.
 func PhysicalClusterNamespaceName(l NamespaceLocator) (string, error) {
 	b, err := json.Marshal(l)
 	if err != nil {
 		return "", err
 	}
-	hash := sha256.Sum224(b)
-	return fmt.Sprintf("kcp%x", hash), nil
+	// Append the logicalCluster marshaled bytes with the workloadClusterName to get a hashed value
```

Comment:
What does this comment mean ... ? The hash is not of the logical cluster, and there is no appending happening anywhere

Reply:
This was a leftover comment. fixed.